pkgs/build-support/setup-hooks/writable-tmpdir-as-home.sh

```diff
@@ -0,0 +1,14 @@
+# shellcheck shell=bash
+# This setup hook set the HOME environment variable to a temporary directory.
+
+export HOME
+
+writableTmpDirAsHome () {
+    if [[ ! -w "$HOME" ]]; then
+        HOME=$(mktemp -d "$NIX_BUILD_TOP/home.XXXX")
```

Comment:
Will `mktemp -d` has consistent dirname in two version of build?

Reply:
This is actually a good point too, updated.